packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelConfigurationController.java

```diff
@@ -27,137 +27,245 @@
 import org.springframework.web.bind.annotation.RestController;
 import org.springframework.web.server.ResponseStatusException;
 import software.uncharted.terarium.hmiserver.models.dataservice.ResponseDeleted;
-import software.uncharted.terarium.hmiserver.models.dataservice.model.ModelConfiguration;
+import software.uncharted.terarium.hmiserver.models.dataservice.model.Model;
+import software.uncharted.terarium.hmiserver.models.dataservice.model.configurations.InitialSemantic;
+import software.uncharted.terarium.hmiserver.models.dataservice.model.configurations.ModelConfiguration;
+import software.uncharted.terarium.hmiserver.models.dataservice.model.configurations.ObservableSemantic;
+import software.uncharted.terarium.hmiserver.models.dataservice.model.configurations.ParameterSemantic;
 import software.uncharted.terarium.hmiserver.security.Roles;
 import software.uncharted.terarium.hmiserver.service.CurrentUserService;
 import software.uncharted.terarium.hmiserver.service.data.ModelConfigurationService;
+import software.uncharted.terarium.hmiserver.service.data.ModelService;
 import software.uncharted.terarium.hmiserver.service.data.ProjectService;
-import software.uncharted.terarium.hmiserver.utils.rebac.Schema;
+import software.uncharted.terarium.hmiserver.utils.Messages;
+import software.uncharted.terarium.hmiserver.utils.rebac.Schema.Permission;
 
 @RequestMapping("/model-configurations")
 @RestController
 @RequiredArgsConstructor
 @Slf4j
-@Transactional
 public class ModelConfigurationController {
```

Comment:
@dvince2 So some observations:

### In general
Can you provide some examples of how CRUD would work with the new data structure and API-end points?


#### posting a new model-config
It seems like when creating a new model-config, we need to pass in the list-form of the Semantics, initailSemanticList, parameterSemanticList ... etc. This isn't readily available in the UI, because the payload had been transformed to the map-version under ModelConfiguration.values. 

So posting this worked:
```
{
  "name": "Default Name",
  "description": "Default Description",
  "fileNames": [
    "string"
  ],
  "temporary": true,
  "publicAsset": true,
  "calibrationRunId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "modelId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "initialSemanticList": [
    {
      "source": "string",
      "type": "initial",
      "target": "abc",
      "expression": "3.0"
    }
  ]
}
```

but something like this fails

```
{
  "name": "Default Name",
  "description": "Default Description",
  "fileNames": [
    "string"
  ],
  "temporary": true,
  "publicAsset": true,
  "calibrationRunId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "modelId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "values": {
    "abc": {
      "source": "string",
      "type": "initial",
      "target": "abc",
      "expression": "3.0"
    }
  }
}
```

#### get
I can't seem to GET model-configurations, they result in 401s

Reply:
I've resolved the weirdness around the lists/maps... there are now just the lists and ive confirmed they're working.

Unfortunately like most of the other API endpoints you ned to provide a project ID you have permissions to